tests/unit/files/files-manager.test.ts

```diff
@@ -0,0 +1,409 @@
+import { FilesManager } from '../../../src/files';
+import { FileResponse } from '../../../src/files/types';
+import { HttpClient } from '../../../src/http';
+import { MockHttpClient } from '../mocks';
+
+describe('FilesManager', () => {
+  let httpClient: HttpClient;
+  let filesManager: FilesManager;
+  let mockFetch: jest.Mock;
+
+  // Mock file data for testing
+  const mockFile: FileResponse = {
```

Comment:
Could we move those example files to the fixtures so that they can be reused by other tests? (+ make the test itself more readable)